Sources/Brave/Frontend/Browser/Search/Voice Search/SpeechRecognizer.swift

```diff
@@ -0,0 +1,300 @@
+            if let formattedTranscript = result?.bestTranscription.formattedString, !formattedTranscript.isEmpty {
+              self.finalize(searchQuery: formattedTranscript)
+            }
+          }
+        }
+      })
+    } catch {
+      reset()
+      log.debug("Voice Search Recognization Fault \(error.localizedDescription)")
+    }
+  }
+    
+  /// Reset the speech recognizer.
+  private func reset() {
+    try? AVAudioSession.sharedInstance().setActive(false)
+    task?.cancel()
+    audioEngine?.stop()
+    
+    audioEngine = nil
+    request = nil
+    task = nil
+  }
+    
+  private func setupStartEngine() throws -> (AVAudioEngine, SFSpeechAudioBufferRecognitionRequest) {
+    let audioEngine = AVAudioEngine()
+    
+    let request = SFSpeechAudioBufferRecognitionRequest()
+    request.shouldReportPartialResults = true
+    request.requiresOnDeviceRecognition = true
+
+    let audioSession = AVAudioSession.sharedInstance()
+    try audioSession.setCategory(.playAndRecord, mode: .measurement, options: .duckOthers)
+    try audioSession.setActive(true, options: .notifyOthersOnDeactivation)
+    let inputNode = audioEngine.inputNode
+    
+    let recordingFormat = inputNode.outputFormat(forBus: 0)
+    inputNode.installTap(onBus: 0, bufferSize: 1024, format: recordingFormat) { [weak self] buffer, when in
+      guard let self else { return }
+      
+      request.append(buffer)
+      
+      guard let channelData = buffer.floatChannelData?[0] else {
+        return
+      }
+      
+      let volume = self.getVolumeLevel(from: channelData)
+      
+      Task { @MainActor in
+        self.setupAnimationWithVolume(volume)
+      }
+    }
+    
+    audioEngine.prepare()
+    try audioEngine.start()
+    
+    return (audioEngine, request)
+  }
+
+  nonisolated private func transcribe(_ message: String) {
+    Task { @MainActor in
+      transcript = message
+    }
+  }
+  
+  nonisolated private func finalize(searchQuery: String) {
+    Task { @MainActor in
+      if !finalizedRecognition.status {
+        finalizedRecognition = (true, searchQuery)
+      }
+    }
+  }
+  
+  nonisolated private func clearSearch() {
+    Task { @MainActor in
+      finalizedRecognition = (false, "")
+    }
+  }
+  
+  private func getVolumeLevel(from channelData: UnsafeMutablePointer<Float>) -> Float {
+    let channelDataArray = Array(UnsafeBufferPointer(start: channelData, count: 1024))
+
+    var outEnvelope = [Float]()
+    var envelopeState: Float = 0
+    let envConstantAtk: Float = 0.16
+    let envConstantDec: Float = 0.003
+
+    for sample in channelDataArray {
+      let rectified = abs(sample)
+
+      if envelopeState < rectified {
+        envelopeState += envConstantAtk * (rectified - envelopeState)
+      } else {
+        envelopeState += envConstantDec * (rectified - envelopeState)
+      }
+      outEnvelope.append(envelopeState)
+    }
+
+    // 0.007 is the low pass filter to prevent
```

Comment:
This number does not match what's in code below

Reply:
Was testing low pass filter as 0.007. It is fixed and proper comments are added.

Fix at https://github.com/brave/brave-ios/pull/7551/commits/9919db2ea0f5f45e36ab3fba198c73aba96ac31a